### ISSUE TITLE
🐛 Semantics: Incorrect redeclaration error for parameters of arrow function expressions

### ISSUE THREAD

**Issue by Boshen** [rome/tools#2960]
### Environment information

[Playground](https://play.rome.tools/?code=const+%7B%0A++a+%3D+%28arg%29+%3D%3E+%7B%7D%2C%0A++b+%3D+%28arg%29+%3D%3E+%7B%7D%2C%0A%7D+%3D+props%3B&lineWidth=80&indentStyle=tab&quoteStyle=double&indentWidth=2&sourceType=module&typescript=true&jsx=true#YwBvAG4AcwB0ACAAewAKACAAIABhACAAPQAgACgAYQByAGcAKQAgAD0APgAgAHsAfQAsAAoAIAAgAGIAIAA9ACAAKABhAHIAZwApACAAPQA+ACAAewB9ACwACgB9ACAAPQAgAHAAcgBvAHAAcwA7AA==)

This is found inside [material-ui](https://github.com/mui/material-ui/blob/master/packages/mui-base/src/AutocompleteUnstyled/useAutocomplete.js).

### What happened?

```
const {
  a = (arg) => {},
  b = (arg) => {},
} = props;
```

errored with
```
error[SyntaxError]: Declarations inside of a `const` declaration may not have duplicates
  ┌─ main.js:3:8
  │
2 │   a = (arg) => {},
  │        --- `arg` is first declared here
3 │   b = (arg) => {},
  │        ^^^ a second declaration of `arg` is not allowed
```

### Expected result

No error.

### Code of Conduct

- [X] I agree to follow Rome's Code of Conduct

**Comment by Boshen** [comment]
Found another case in [react-admin](https://github.com/marmelab/react-admin/blob/master/packages/ra-ui-materialui/src/layout/Menu.tsx#L53-L54)

```
error[SyntaxError]: Declarations inside of a `const` declaration may not have duplicates
   ┌─ react-admin/packages/ra-ui-materialui/src/layout/Menu.tsx:54:22
   │
53 │                 .filter(name => resources[name].hasList)
   │                         ---- `name` is first declared here
54 │                 .map(name => (
   │                      ^^^^ a second declaration of `name` is not allowed
```

**Comment by IWANABETHATGUY** [comment]
For this checking, 
## Interface
https://www.typescriptlang.org/play?ssl=22&ssc=1&pln=1&pc=1#code/JYOwLgpgTgZghgYwgAgM5iqA5sg3gXwChDRJZEUQBXAGxryJPGniWWoFsAjaB40lhWQB7LgCsICMHyZlWKOCACeMgeTZUQAaxDCA7iFXN1KLsOE0Iio3KFdgWUjcFtUS7hecnkAN2HAAEy95dggfXgJCIA
**necessary checking**

## Type Alias
https://www.typescriptlang.org/play?ssl=22&ssc=1&pln=1&pc=1#code/C4TwDgpgBAzsBOBLAdgcygXigbwL4Ch9RIpkBXAGwsxwKPGnIFsAjCeGvQ46AexYBWEAMbBOdHlACGyEOO4MoZZAGtkvAO7J59Ei168KEGTsktEqFGKxdd0GCFaHTigG69EAExclkEV+zyQA
**necessasry checking**
## Module
https://www.typescriptlang.org/play?ssl=22&ssc=1&pln=1&pc=1#code/LYewJgrgNgpgBAZwC4CcCWA7A5nA3gXwChDRJY4Noo8iTxp5LgAjGFG40huEZgKxgBjJBzpl4AQwwBPUV3IQMAawwgA7hjn1yzECFhSt4uMzRZMIgp23wE0lvqPcAbiDRgn5DDGdsOQA
for `module` we only need to check `null` and `void`

**Comment by MichaReiser** [comment]
Yeah, moving this to the analyzer is a good idea, now that @xunilrj implemented the semantic model. 

One thing that we may need to do tough is to change the t262 tests to no longer run against the parser only but run with analyze. 

@xunilrj is this something you might be interested in looking at? There are probably more checks that we could move from our parser into the analyzer.

**Comment by xunilrj** [comment]
There are a couple of other changes I want to move from the parser to linters. I will create a small list and we can decide if makes sense or not.

**Comment by ematipico** [comment]
Going to re-open: @xunilrj are there any progress around this issue? Should we close it?
